### DISCUSSION TITLE
MacOS app keeps asking for permissions that it has already been granted

### DISCUSSION THREAD

**Discussion by PocketUmby** [whs/runekit#33] [Q&A macOS] [answered]
I was using Runekit on my mac previously, and it worked fine.  I got rid of the app when I stopped playing Runescape over the holidays.  I redownloaded the app when I picked the game back up and proceeded with the same installation instructions that worked previously.  Whenever I open Runekit now though, it asks for accessibility permissions that it has already been granted, then springs in the dock until it goes not responding.  It never goes into a system tray icon like it did previously. It has Accessibility, Screen Recording and Input Monitoring.

I tried `sudo xattr -dr com.apple.quarantine ` and dropping the app into terminal to no avail.

On an M1 Air running macOS 13.3

Any thoughts?

**Comment by PocketUmby** [comment, marked as answer]
Finally fixed this, was a macOS issue where it says that it has given permissions but it actually hasn't.  

Running

`sudo tccutil reset Accessibility` 
followed by 
`sudo tccutil reset ScreenCapture` 
and then opening RuneKit allowed it to request the permissions properly again, and when given them this time it worked.  Note this will reset accessibility and screen capture settings for all apps so you might get some more prompts from discord and the like.

**Comment by nayfaan** [comment]
Hi, @PocketUmby, I tried your fix but the problem still persist on my computer sadly ): Any ideas why it could be the case?

**Reply by nayfaan** [reply]
#38 
Dug around a bit. It turns out that it's runekit not realizing that it has the permissions already despite actually having the permission
